java-frontend/src/main/java/org/sonar/java/resolve/TypeSubstitution.java

```diff
@@ -28,6 +28,7 @@
 import java.util.Map;
 
 public class TypeSubstitution {
+  public boolean unchecked = false;
```

Comment:
I also agree with that issue raised by SonarQube. I would prefer to have this private, and possibly final... Only visible though a getter `isUnchecked()`. 

The only case where you need to set it to "true" is when identifying raw type or null type in `TypeInferenceSolver`. It is then never modified. I would prefer to have it explicitly in `TypeSubstitution`. 

I would propose two approaches:

* Having a dedicated method doing only that, allowing to have only `TypeSubstitution` knowing how to build a unchecked `TypeSubstitution`. Which is probably a bit more elegant.
```java
public static SubstitutionType uncheckedTypeSubstitution() {
  TypeSubstitution typeSubstitution = new TypeSubstitution();
  typeSubstitution.unchecked = true;
  return typeSubstitution;
}
```

* Or going further and reuse a constant everytime we need a unchecked TypeSubstitution:
```java
class TypeSubstitution {

  private static final UNCHECKED = new TypeSubstitution() {
    @Override
    public boolean isUnchecked() {
      return true;
    }
  }

  public boolean isUnchecked() {
    return false;
  }

  public static SubstitutionType uncheckedTypeSubstitution() {
    return UNCHECKED;
  }
}
```